src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
+    case value
+    when "", "NULL", "Null", "null", "~"
+      "null"
+    when "true", "True", "TRUE", "on", "On", "ON", "y", "Y", "yes", "Yes", "YES"
+      "true"
+    when "false", "False", "FALSE", "off", "Off", "OFF", "n", "N", "no", "No", "NO"
+      "false"
+    when ".inf", ".Inf", ".INF", "+.inf", "+.Inf", "+.INF"
+      "infinity"
+    when "-.inf", "-.Inf", "-.INF"
+      "-infinity"
+    when ".nan", ".NaN", ".NAN"
+      "NaN"
+    else
+      if value.to_i64?
```

Comment:
Yes, you must specify `to_i?(whitespace: false, underscore: false)` and keep `strict` and `prefix` enabled, otherwise integers and floats with leading/suffix spaces or underscores that are valid Crystal numbers would get parsed, even though they're invalid YAML numbers.